src/rapidata/rapidata_client/order/rapidata_order_builder.py

```diff
@@ -143,11 +143,12 @@ def create(self, submit: bool = True, max_workers: int = 10) -> RapidataOrder:
         )
 
         self.order_id = result.order_id
-        self._dataset = RapidataDataset(result.dataset_id, self._openapi_service)
+        self._dataset = RapidataDataset(result.dataset_id, self._openapi_service) # type: ignore - should only be None if order is evaluation order
```

Comment:
Would something like this be cleaner than a type ignore?

```suggestion
        self._dataset = RapidataDataset(result.dataset_id, self._openapi_service)  if result.dataset_id else None
```